TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/integ_test/integ_test/backfill_tests.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import pytest
+import unittest
+from http import HTTPStatus
+from console_link.logic.clusters import connection_check, clear_indices, run_test_benchmarks, ConnectionResult
+from console_link.models.cluster import Cluster
+from console_link.models.backfill_base import Backfill
+from console_link.cli import Context
+from common_operations import (get_document, create_document, create_index, check_doc_counts_match,
+                               EXPECTED_BENCHMARK_DOCS)
+
+logger = logging.getLogger(__name__)
+
+
+def preload_data(source_cluster: Cluster, target_cluster: Cluster):
+    # Confirm source and target connection
+    source_con_result: ConnectionResult = connection_check(source_cluster)
+    assert source_con_result.connection_established is True
+    target_con_result: ConnectionResult = connection_check(target_cluster)
+    assert target_con_result.connection_established is True
+
+    # Clear any existing non-system indices
+    clear_indices(source_cluster)
+    clear_indices(target_cluster)
+
+    # Preload data that test cases will verify is migrated
+    # test_backfill_0001
+    index_name = f"test_backfill_0001_{pytest.unique_id}"
+    doc_id = "backfill_0001_doc"
+    create_index(cluster=source_cluster, index_name=index_name)
+    create_document(cluster=source_cluster, index_name=index_name, doc_id=doc_id,
+                    expected_status_code=HTTPStatus.CREATED)
+
+    # test_backfill_0002
+    run_test_benchmarks(source_cluster)
+
+
+@pytest.fixture(scope="class")
+def setup_backfill(request):
+    config_path = request.config.getoption("--config_file_path")
+    unique_id = request.config.getoption("--unique_id")
+    pytest.console_env = Context(config_path).env
+    pytest.unique_id = unique_id
+    preload_data(source_cluster=pytest.console_env.source_cluster,
+                 target_cluster=pytest.console_env.target_cluster)
+    backfill: Backfill = pytest.console_env.backfill
+    assert backfill is not None
+    backfill.create()
+    backfill.start()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cleanup_after_tests():
+    # Setup code
+    logger.info("Starting backfill tests...")
+
+    yield
+
+    # Teardown code
+    logger.info("Stopping backfill...")
+    backfill: Backfill = pytest.console_env.backfill
+    backfill.stop()
+
+
+@pytest.mark.usefixtures("setup_backfill")
+class BackfillTests(unittest.TestCase):
+
+    def test_backfill_0001_single_document(self):
```

Comment:
I typically expect to set tests look like 1) Setup, 2) Action, 3) Assertions; I'm having a hard time seeing how these tests work, as it seems like these are only doing 3) Assertions.  Maybe you could help me understand?

Reply:
Let me add some documentation around this. I do expect to have tests like you mentioned, this test class does this as well but in a grouped manner to allow starting and stopping a backfill service a single time for more basic test cases, such that the test cases themselves are only validating the backfill with another method preloading the data for the test cases

Reply:
Lets roll with what we have, while I'd like for it to be more approachable maybe we can address this over time rather than forcing it right here/now.